SpatialGDK/Source/SpatialGDKFunctionalTests/Public/TestMaps/GeneratedTestMap.h

```diff
@@ -3,41 +3,47 @@
 #pragma once
 
 #include "CoreMinimal.h"
+
+#include "GeneratableTestMap.h"
 #include "GenerateTestMapsCommandlet.h"
 #include "GeneratedTestMap.generated.h"
 
+enum class EMapCategory
+{
+	CI_PREMERGE,
+	CI_PREMERGE_SPATIAL_ONLY,
+	CI_NIGHTLY,
+	CI_NIGHTLY_SPATIAL_ONLY,
+	NO_CI
+};
+
 /**
  * The base class for automatically generating test maps.
  * Descend from this and your class should automatically be picked up by the GenerateTestMapsCommandlet for generation.
  */
-UCLASS(Abstract)
-class SPATIALGDKFUNCTIONALTESTS_API UGeneratedTestMap : public UObject
+UCLASS()
+class SPATIALGDKFUNCTIONALTESTS_API UGeneratedTestMap : public UObject, public IGeneratableTestMap
 {
 	GENERATED_BODY()
 
 public:
 	UGeneratedTestMap();
-	bool GenerateMap();
-	bool GenerateCustomConfig();
-	virtual bool ShouldGenerateMap() { return true; } // To control whether to generate a map from this class
-	FString GetMapName() { return MapName; }
-	static FString GetGeneratedMapFolder();
-
-protected:
-	enum class EMapCategory
-	{
-		CI_PREMERGE,
-		CI_PREMERGE_SPATIAL_ONLY,
-		CI_NIGHTLY,
-		CI_NIGHTLY_SPATIAL_ONLY,
-		NO_CI
-	};
 
 	// This constructor with arguments is the one that the descended classes should be using within their no-argument constructor.
 	UGeneratedTestMap(EMapCategory MapCategory, FString MapName);
 
-	// This is what test maps descending from this class should normally override to add their own content to the map
-	virtual void CreateCustomContentForMap() {}
+	// Should be used directly after the no argument constructor.
+	void Init(EMapCategory InMapCategory, FString InMapName);
+
+	void GenerateMap() override;
```

Comment:
mixture of virtual and non-virtual specifiers?

Reply:
+1 I am personally a fan of omitting `virtual` when you specify `override` but I don't think we have a hard and fast rule about it yet. In any case, being consistent reads a bit easier.

Reply:
? override is overrideable? I would pick one style you agree with and apply it everywhere. If you don't want virtual method to be overridden use final instead.

Reply:
ok yeah, the opposite way round with final makes much more sense, though maybe not here because it would be on nearly everything. Removed the virtual.

Thanks for bearing with me 😄